File: presentation/src/main/java/com/example/presentation/ui/MainActivity.java
package com.example.presentation.ui;

import androidx.annotation.NonNull;
import androidx.annotation.Nullable;
import androidx.appcompat.app.AppCompatActivity;
import androidx.fragment.app.Fragment;
import androidx.fragment.app.FragmentManager;
import androidx.fragment.app.FragmentStatePagerAdapter;
import androidx.viewpager.widget.PagerAdapter;
import androidx.viewpager.widget.ViewPager;

import android.content.Context;
import android.content.res.Resources;
import android.os.Bundle;

import com.example.presentation.R;
import com.example.presentation.utils.DateUtils;

public class MainActivity extends AppCompatActivity {


    private ViewPager mPager;
    private PagerAdapter mPagerAdapter;

    @Override
    protected void onCreate(Bundle savedInstanceState) {
        super.onCreate(savedInstanceState);
        setContentView(R.layout.activity_main);

        mPager = findViewById(R.id.pager);
        mPagerAdapter = new PageAdapter(getSupportFragmentManager(), 0, this);
        mPager.setAdapter(mPagerAdapter);
    }

    private static class PageAdapter extends FragmentStatePagerAdapter {

        private final Resources mResources;

        PageAdapter(@NonNull FragmentManager fm, int behavior, Context context) {
            super(fm, behavior);
            mResources = context.getResources();
        }

        @NonNull
        @Override
        public Fragment getItem(int position) {
            return MainFragment.newInstance(position);
        }

        @Override
        public int getCount() {
            return DateUtils.getLengthOfYear();
        }

        @Nullable
        @Override
        public CharSequence getPageTitle(int position) {
            return mResources.getString(R.string.astronomy_picture, DateUtils.getDateOffset(position));

        }
    }
}

File: presentation/src/main/java/com/example/presentation/ui/MainActivity.kt
package com.example.presentation.ui

import android.content.Context
import android.content.res.Resources
import android.os.Bundle
import androidx.appcompat.app.AppCompatActivity
import androidx.fragment.app.FragmentManager
import androidx.fragment.app.FragmentStatePagerAdapter
import androidx.viewpager.widget.PagerAdapter
import androidx.viewpager.widget.ViewPager
import com.example.presentation.R
import com.example.presentation.ui.fragment.ImageFragment
import com.example.presentation.ui.fragment.MainFragment
import com.example.presentation.utils.DateUtils.getDateOffset
import com.example.presentation.utils.DateUtils.lengthOfYear

class MainActivity : AppCompatActivity(), ZoomClickListener {

    override fun onCreate(savedInstanceState: Bundle?) {
        super.onCreate(savedInstanceState)
        setContentView(R.layout.activity_main)
        val pager = findViewById<ViewPager>(R.id.pager)
        val pagerAdapter: PagerAdapter = PageAdapter(supportFragmentManager, 0, this)
        pager.adapter = pagerAdapter
    }

    override fun onZoomImage(url: String) {
        val imageFragment = ImageFragment.newInstance(url)
        supportFragmentManager
                .beginTransaction()
                .addToBackStack(null)
                .replace(R.id.root_view, imageFragment)
                .commit()
    }

    private class PageAdapter internal constructor(fm: FragmentManager, behavior: Int, context: Context) : FragmentStatePagerAdapter(fm, behavior) {
        private val mResources: Resources = context.resources

        override fun getItem(position: Int) = MainFragment.newInstance(position)

        override fun getCount() = lengthOfYear()


        override fun getPageTitle(position: Int) = mResources.getString(R.string.astronomy_picture, getDateOffset(position))


    }
}